src/d3d9/d3d9_device.cpp

```diff
@@ -427,7 +427,7 @@ namespace dxvk {
       return D3DERR_INVALIDCALL;
 
     try {
-      const Com<D3D9Texture2D> texture = new D3D9Texture2D(this, &desc);
```

Comment:
pSharedHandle can be a regular void Pointer here. Basically just move up the if in line 434 and use that to decide whether or not you pass the handle to the constructor.

> In Windows Vista CreateTexture can create a texture from a system memory pointer allowing the application more flexibility over the use, allocation and deletion of the system memory. For example, an application could pass a GDI system memory bitmap pointer and get a Direct3D texture interface around it. Using a system memory pointer with CreateTexture has the following restrictions.

> The pitch of the texture must be equal to the width multiplied by the number of bytes per pixel.
    Only textures with a single mipmap level are supported. The Levels argument must be 1.
    The Pool argument must be D3DPOOL_SYSTEMMEM.
    The pSharedHandle argument must be a valid pointer to a buffer that can hold the system memory point; *pSharedHandle must be a valid pointer to system memory with a size in bytes of texture width * texture height * bytes per pixel of the texture format.

Reply:
Should be resolved